### PR TITLE
Task/homepage improvement testing

### PR DESCRIPTION
**Pull request by ryanryanchan**
## Description
Updated New Homepage
- speed updates using redis caching for img data
- default images implemented to avoid dead image links
- Popular publications implemented
- admin control page for popular publications implemented
- bugfix: clicking on publications now links to document details page correctly
- bugfix: formatting search Time string in recent searches
- bugfix: long titles have been truncated to 50 characters, at the end of the nearest word. also using Title Case for text.
## Related Issue/Ticket

https://jira.di2e.net/browse/UOT-109748
https://jira.di2e.net/browse/UOT-107944

## Testing instructions

- go to http://10.194.9.112:8080/#/gamechanger-test
- check load times are bearable to the average human
- check loading behavior, default images are used until the actual images load + swap in. 
- click on a source. verify it links to a source details page correctly
- hover over any publication. check to see if the Title Case is used, and if long titles are truncated well. 
- click on a publication. verify that it goes to the details page correctly 
- go to admin -> homepage editor. there you should see a table to enable / disable entries in popular publications. check that both disabling entries + re-enabling entries works as expected (make sure to to hit save after making changes to that table)

## Unit Testing
backend/test/controllers/documentController.test.js
#getHomepageThumbnail
backend/test/utils/searchUtility.test.js
#getPopularQuery

### CODE REVIEW

**Comment by rha930** [comment]
lets switch over to the real mainview

File: frontend/src/components/modules/policy/policyTestMainViewHandler.js [hunk 1]
```diff
@@ -288,7 +349,9 @@ const PolicyMainViewHandler = {
 		} = state;
 
 		const showDidYouMean = didYouMean && !loading;
-		const trendingStorage = localStorage.getItem(`trending${cloneData.clone_name}Searches`) || '[]';
+		// const trendingStorage = localStorage.getItem(`trending${cloneData.clone_name}Searches`) || '[]';
```
Comment (kmanne09) [inline frontend/src/components/modules/policy/policyTestMainViewHandler.js:352]: This can be switched back when moving the homepage to the /#/gamechanger.

**Review by kmanne09 — COMMENTED** [review]
Looks good to me. Agree with Rob, last thing is moving it to the main policy view.

**Review by JLyons1985 — DISMISSED** [review]

File: frontend/src/components/modules/default/defaultMainViewHandler.js [hunk 2]
```diff
@@ -193,8 +193,8 @@ const DefaultMainViewHandler = {
 			// 	const names = data.docs.map(d=>d.crawler_name)
 			// 	setState(dispatch, {crawlerSources: names});
 			// });
-			let crawlerSources = await gameChangerAPI.gcCrawlerSealData();
-			setState(dispatch, {crawlerSources: crawlerSources.data});
+			// let crawlerSources = await gameChangerAPI.gcCrawlerSealData();
```
[the remark this answers is not in the document]
Comment (ryanryanchan) [reply frontend/src/components/modules/default/defaultMainViewHandler.js:196]: good point. removed

File: frontend/src/components/modules/policy/policyTestMainViewHandler.js (hunk 1 above)
Comment (ryanryanchan) [reply frontend/src/components/modules/policy/policyTestMainViewHandler.js:316]: done

**Review by rha930 — APPROVED** [review]

**Review by JLyons1985 — APPROVED** [review]

**Pull request merged**